### PR TITLE
Use fingerprint to check instead content for public key

### LINKED ISSUE #697: Every public key after the first added is a 'duplicate' for MSSQL
- Gitea version (or commit ref): d2bb8ef5039
- Git version: 2.9.1
- Operating system: Windows 10
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist: n/a

## Description

Using mssql results in every key after the first key being detected as already existing.
It appears the check for "does this key exists" is actually running "does any key exist"

`models/ssh_key.go line 360` is the line.

This is most likely caused by go-xorm. It appears as if it is ignoring the text condition for mssql.
A workaround is 
```has, err := x.Where("content=?",content).Where("type=?",KeyTypeUser).Get(&PublicKey{})```

...

### PR DESCRIPTION
**Pull request by lunny**
fix #697

### CODE REVIEW

File: models/ssh_key.go [hunk 1]
```diff
@@ -354,41 +354,50 @@ func appendAuthorizedKeysToFile(keys ...*PublicKey) error {
...
+func checkKeyFingerprint(e Engine, fingerprint string) error {
+	has, err := e.Get(&PublicKey{
+		Fingerprint: fingerprint,
+		Type:        KeyTypeUser,
 	})
 	if err != nil {
 		return err
 	} else if has {
-		return ErrKeyAlreadyExist{0, content}
+		return ErrKeyAlreadyExist{0, fingerprint}
```
Comment (ethantkoenig) [inline models/ssh_key.go]: Could we update the `ErrKeyAlreadyExist` struct to have a `Fingerprint` field instead of `Content`?

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

File: models/ssh_key.go (hunk 1 above)
Comment (lunny) [reply models/ssh_key.go]: done.

**Review by ethantkoenig — APPROVED** [review]

**Comment by ethantkoenig** [comment]
LGTM

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Pull request merged**